stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2475,30 +2516,41 @@ internal class _RawNative${Self}Buffer:
     return UnsafeMutablePointer(Builtin.projectTailElems(self, UInt.self))
   }
 
+  /// The empty singleton that is used for every single Dictionary that is
+  /// created without any elements. The contents of the buffer should never
+  /// be mutated.
+  // FIXME: this should be set up in rodata like the Array singleton
   internal static let empty: RawBuffer = {
     let buffer = Builtin.allocWithTailElems_1(RawBuffer.self,
         1._builtinWordValue, UInt.self)
     
     // We set the capacity to 1 so that there's an empty hole to search.
-    // Any insertion will reallocate because there always needs to be
-    // an empty hole.
+    // Any insertion will lead to a real buffer being allocated, because 
+    // Dictionary guarantees there's always another empty hole after insertion.
     buffer._body = StorageHeader(capacity: 1)
 
     let initializedEntries = _UnsafeBitMap(
         storage: buffer._initializedHashtableEntriesBitMapStorage,
         bitCount: 1)
     initializedEntries.initializeToZero()
 
-    // Initialize header
+    // We don't bother to initalize the pointers in the header because
+    // they should never be accessed on the empty singleton.
     buffer._body.initializedEntries = initializedEntries
     return buffer
   }()
 
   internal init(_doNotCallMe: ()) {
-    _sanityCheckFailure("Only initialize these by calling create")
+    _sanityCheckFailure("Only create this by using the `empty` singleton")
```

Comment:
Why is this init needed?  A comment should say.  Also, this should probably call `Builtin.unreachable` if that's not already part of `_sanityCheckFailure`